homeassistant/components/switch/edimax.py

```diff
@@ -0,0 +1,75 @@
+"""
+homeassistant.components.switch.edimax
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Support for Edimax switches.
+"""
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.const import CONF_HOST, CONF_USERNAME, CONF_PASSWORD
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Find and return Edimax Smart Plugs. """
+    try:
+        # pylint: disable=no-name-in-module, import-error
+        from pyedimax.smartplug import SmartPlug
+    except ImportError:
+        logging.getLogger(__name__).exception((
+            "Failed to import pyedimax. "))
+
+        return
+
+    host = config.get(CONF_HOST)
+    auth = (config.get(CONF_USERNAME, 'admin'),
+            config.get(CONF_PASSWORD, '1234'))
+
+    if not host:
+        logging.getLogger(__name__).error(
+            'Missing config variable %s', CONF_HOST)
```

Comment:
The preferred way of doing this is by using homeassistant.helpers.validate_config